PIL/TiffImagePlugin.py

```diff
@@ -1737,17 +1737,28 @@ def fixOffsets(self, count, isShort=False, isLong=False):
                 self.rewriteLastLong(offset)
 
 def _save_all(im, fp, filename):
-    if not hasattr(im, "n_frames"):
+    encoderinfo = im.encoderinfo.copy()
+    encoderconfig = im.encoderconfig
+    append_images = encoderinfo.get("append_images", [])
+    if not hasattr(im, "n_frames") and not len(append_images):
         return _save(im, fp, filename)
 
     cur_idx = im.tell()
     try:
         with AppendingTiffWriter(fp) as tf:
-            for idx in range(im.n_frames):
-                im.seek(idx)
-                im.load()
-                _save(im, tf, filename)
-                tf.newFrame()
+            for ims in [im]+append_images:
+                ims.encoderinfo = encoderinfo
+                ims.encoderconfig = encoderconfig
+                if not hasattr(ims, "n_frames"):
+                    nfr = 1
+                else:
+                    nfr = ims.n_frames
```

Comment:
The best would be to be able to append images with different specifications as well, but even this is better than the current situation, when one can save a multiframe tiff only from another multiframe image. That's why I wrote "a quick solution" in the commit message.